lib/psci/psci_main.c

```diff
@@ -44,6 +44,23 @@ int psci_cpu_on(u_register_t target_cpu,
 	return psci_cpu_on_start(target_cpu, &ep);
 }
 
+int psci_platform_migrate_info(u_register_t *mpidr)
```

Comment:
@ldts, as I commented earlier, these generic code changes need to be pulled out as a separate PR and we can discuss as how to implement your requirement for preventing CPU_OFF on primary CPU.